en/02_Developer_Guides/04_Configuration/00_Configuration.md

```diff
@@ -24,175 +24,163 @@ For providing content editors or CMS users a place to manage configuration see t
+If you want a private static property that has no interactions with the configuration API, you can
+mark it `@internal` in the property's PHPdoc.
 
+```php
+/**
+ * @internal
+ */
+private static $not_config;
 ```
+[/info]
 
-## Accessing and Setting Configuration Properties
+## Accessing Configuration Properties
 
 This can be done by calling the static method [Config::inst()](api:SilverStripe\Core\Config\Config::inst()), like so:
 
-
 ```php
-$config = Config::inst()->get('MyClass', 'property');
+$config = Config::inst()->get(MyClass::class, 'property');
 ```
 
-Or through the `config()` method on the class.
+Or through the [`config()`](api:SilverStripe\Core\Config\Configurable::config()) method on the class (assuming the class uses the [`Configurable`](api:SilverStripe\Core\Config\Configurable) trait).
 
 ```php
-$config = $this->config()->get('property');
+$config = MyClass::config()->get('property');
+$config = static::config()->get('property');
 ```
 
-You may need to apply the [Configurable](api:SilverStripe\Core\Config\Configurable) trait in order to access the `config()` method.
+Note that while both objects have similar methods the APIs differ slightly. The below actions are equivalent:
+
+* `Config::inst()->get(MyClass::class, 'property');` or `MyClass::config()->get('property')` - gets the value of the configuration property whether it was set on this class or one of its ancestors
+* `Config::inst()->uninherited(MyClass::class, 'property');` or `MyClass::config()->get('property', Config::UNINHERITED)` - gets the value of the configuration property only if it was defined for the specific class passed in
 
-**app/src/MyOtherClass.php**
+You can also check whether a class has a value for a configuration property using the following syntax:
 
 ```php
-use SilverStripe\Core\Config\Configurable;
-
-class MyOtherClass 
-{
-    use Configurable;
- 
-}
+Config::inst()->exists(MyClass::class, 'property');
 ```
 
+## Setting Configuration Properties
+
+### At runtime
 
-Note that by default `Config::inst()` returns only an immutable version of config. Use `Config::modify()`
+Note that by default `Config::inst()` returns only an immutable version of config. Use [`Config::modify()`](api:SilverStripe\Core\Config\Config::modify())
 if it's necessary to alter class config. This is generally undesirable in most applications, as modification
 of the config can immediately have performance implications, so this should be used sparingly, or
 during testing to modify state.
 
-Note that while both objects have similar methods the APIs differ slightly. The below actions are equivalent:
+If you do need to modify configuration at run time, you can do so using the following API:
+
+* `Config::modify()->merge(MyClass::class, 'property', 'newvalue');` or `MyClass::config()->merge('property', 'newvalue')` - merges the new configuration value in with any pre-existing values for this property
```

Comment:
I don't think we should be including both ways of doing them when they're equivalent. 

While it's OK to mention at the start there are 2x different way so to do it, `MyClass::config()->merge('property', 'newvalue')` is probably the better of the two since it's shorter, and reinforce s the fact you need to have the `Configurable` trait applied to the class in order for config to work.

Therefore could you please remove any references to `Config::modify()->merge(MyClass::class, 'property', 'newvalue');` style setting here and in the rest of this doc

Reply:
This goes into "best practices" which will be covered in https://github.com/silverstripe/developer-docs/issues/269 later on. For now, both of these are valid ways to access the config and both work just fine. The scope of this PR and the issue it relates to is primarily "are the docs factually correct?"
See https://github.com/silverstripe/developer-docs/issues/212 regarding scope.

Reply:
Also, config _does_ work for classes that don't have the `Configurable` trait. It's _recommended_ to have that trait, but it's not a requirement.